From 6b435d36ef318c27dbd04bf3ad12eb771cb04ede Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Tue, 17 Dec 2019 10:55:31 +0100
Subject: [PATCH 02/97] can: m_can: Set pinmux into "default" state on start

On systems like stm32mp1 where pins have both "default" and "sleep" pinmux
states in DT, the pins are in "sleep" state by default. Switch the pins into
the "default" state when bringing the interface up in m_can_start(), because
otherwise no CAN communication is possible. This replicates the behavior of
the resume path, which does the same.

Signed-off-by: Marek Vasut <marex@denx.de>
Cc: Alexandre Torgue <alexandre.torgue@st.com>
Cc: Marc Kleine-Budde <mkl@pengutronix.de>
Cc: Wolfgang Grandegger <wg@grandegger.com>
To: linux-can@vger.kernel.org
---
 drivers/net/can/m_can/m_can.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/can/m_can/m_can.c b/drivers/net/can/m_can/m_can.c
index 562c8317e3aa8..13f3701604120 100644
--- a/drivers/net/can/m_can/m_can.c
+++ b/drivers/net/can/m_can/m_can.c
@@ -1200,6 +1200,8 @@ static void m_can_start(struct net_device *dev)
 	/* basic m_can configuration */
 	m_can_chip_config(dev);
 
+	pinctrl_pm_select_default_state(cdev->dev);
+
 	cdev->can.state = CAN_STATE_ERROR_ACTIVE;
 
 	m_can_enable_all_interrupts(cdev);
-- 
2.27.0

